husky_directory/services/search.py

```diff
@@ -81,17 +82,26 @@ def _translate_pws_list_persons_output(
 
     def search_directory(
         self, request_input: SearchDirectoryInput
-    ) -> Dict[str, SearchDirectoryOutput]:
+    ) -> SearchDirectoryOutput:
         """The main interface for this service. Submits a query to PWS, filters and translates the output,
-        and returns a SearchDirectoryOutput."""
+        and returns a DirectoryQueryScenarioOutput."""
 
-        all_results = {}
+        scenarios: List[DirectoryQueryScenarioOutput] = []
+        num_results = 0
 
         for query_description, query in self.query_generator.generate(request_input):
             pws_output = self._pws.list_persons(query)
             results = self._translate_pws_list_persons_output(pws_output)
             while pws_output.next and pws_output.next.href:
                 pws_output = self._pws.get_next(pws_output.next.href)
                 results.extend(self._translate_pws_list_persons_output(pws_output))
-            all_results[query_description] = SearchDirectoryOutput(people=results)
-        return all_results
+            scenarios.append(
+                DirectoryQueryScenarioOutput(
+                    description=query_description, people=results
+                )
+            )
+            num_results += len(results)
```

Comment:
line 103
Does this give the count data for...
__Last name begins with "wert": 1 Faculty/Staff
Name contains "wert": 1 Faculty/Staff__ ?

Reply:
No, for that you'd still use length of each respective list.

Reply:
(which is still inaccurate until I fix the dedupe problem, which is next on my list)

Reply:
>No, for that you'd still use length of each respective list. There is no other way (without iterating and summing each list) to get the total result count for the query across all scenarios.

Ok, that's what I thought. What is num_results?

Reply:
> the total result count for the query across all scenarios.

Reply:
`SearchDirectoryOutput.num_results` i think tells a pretty clear story?

Reply:
if you think there's a clearer name for this attribute, i'm happy to change i

Reply:
Ok, makes sense it will make even more sense when I get back into the code.

Reply:
ok. FWIW there is nothing in the current feature set requiring the total count, it just seemed like a nice thing to have since otherwise it's a heavy iteration to get that information.

Reply:
It is nice to have, it would tell me for sure if there are any results :)